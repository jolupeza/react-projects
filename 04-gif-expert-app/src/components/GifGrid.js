import GifGridItem from "./GiftGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{ category }</h3>
      { loading && <p className="">Loading...</p> }
      <div className="card-grid">
        {
          images.map( image => <GifGridItem key={image.id} {...image} />)
        }
      </div>
    </>
  );
}

export default GifGrid;
