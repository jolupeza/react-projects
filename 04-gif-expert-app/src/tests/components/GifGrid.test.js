import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test component AddCategory', () => {
  const category = 'Dragon Ball';

  test('should show component GifGrid', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when load images', () => {
    const gifs = [
      { id: 'ABC', url: 'https://localhost/image.png', title: 'Goku'},
      { id: 'DEF', url: 'https://localhost/image.png', title: 'Goku'},
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
