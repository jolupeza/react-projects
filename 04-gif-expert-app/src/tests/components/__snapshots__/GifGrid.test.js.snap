// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component AddCategory should show component GifGrid 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
